### ISSUE TITLE
Intermittent config shape errors during table generation

### ISSUE THREAD

**Issue by Westbrook** [Polymer/tachometer#204]
I've just started getting the following error when attempting to generate tachometer results for my project:

```
config {
  border: {
    topBody: '─',
    topJoin: '┬',
    topLeft: '┌',
    topRight: '┐',
    bottomBody: '─',
    bottomJoin: '┴',
    bottomLeft: '└',
    bottomRight: '┘',
    bodyLeft: '│',
    bodyRight: '│',
    bodyJoin: '│',
    joinBody: '─',
    joinLeft: '├',
    joinRight: '┤',
    joinJoin: '┼'
  },
  columns: [ { alignment: 'right' }, { alignment: 'left' } ]
}
errors [
  {
    dataPath: '/columns',
    message: 'should be object',
    params: { type: 'object' },
    schemaPath: '#/type'
  }
]
Error: Invalid config.
    at validateConfig (/Users/westbrook/Documents/web/repos/swc2/node_modules/tachometer/node_modules/table/src/validateConfig.js:26:11)
    at makeConfig (/Users/westbrook/Documents/web/repos/swc2/node_modules/tachometer/node_modules/table/src/makeConfig.js:55:3)
    at Object.table (/Users/westbrook/Documents/web/repos/swc2/node_modules/tachometer/node_modules/table/src/table.js:83:18)
    at Object.horizontalTermResultTable (/Users/westbrook/Documents/web/repos/swc2/node_modules/tachometer/src/format.ts:152:16)
    at Runner.outputResults (/Users/westbrook/Documents/web/repos/swc2/node_modules/tachometer/src/runner.ts:345:17)
    at Runner.run (/Users/westbrook/Documents/web/repos/swc2/node_modules/tachometer/src/runner.ts:87:16)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
    at realMain (/Users/westbrook/Documents/web/repos/swc2/node_modules/tachometer/src/cli.ts:159:14)
    at Object.main (/Users/westbrook/Documents/web/repos/swc2/node_modules/tachometer/src/cli.ts:40:15)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

I've tracked it back to the following code paths:

https://github.com/Polymer/tachometer/blob/a1c6457048f7511b57995ba0f5cc6ed719195adf/src/format.ts#L118-L129

and:

https://github.com/Polymer/tachometer/blob/a1c6457048f7511b57995ba0f5cc6ed719195adf/src/format.ts#L140-L156

Locally when I move from `.map()` to `.reduce()` and accumulate a number indexed object the situation resolves itself, a la:

```
    const columns = dimensions.reduce((acc, d, i) => {
        acc[i] = d.tableConfig || {};
        return acc;
    }, {});
```

and:

```
    const columns = {
        0: { alignment: 'right' },
        ...results.reduce((acc, _column, i) => {
            acc[i + 1] = { alignment: 'left' };
            return acc;
        }, {}),
    };
```

Does that seem like a reasonable PR?

**Comment by aomarks** [comment]
Thanks for catching! A PR would be great if you have time.

It looks like the cause is a recent regression in the `table` library. I filed https://github.com/gajus/table/issues/134, could you also mention that in a comment near the fix?

The `reduce` approach is fine, though I think I might prefer using something like `Object.fromEntries(dimensions.map((d, i) => [i, ...]))` -- a little more readable I think? You'll just need to bump `es2017` to `es2019` in `tsconfig-base.json` first.

**Comment by aomarks** [comment]
Fix released in 0.5.6

**Comment by Westbrook** [comment]
Thanks @aomarks for beating me to this!
